kubernetes-gcp-go/main.go

```diff
@@ -0,0 +1,152 @@
+			},
+			Location: pulumi.String(gcpRegion),
+			MasterAuthorizedNetworksConfig: &container.ClusterMasterAuthorizedNetworksConfigArgs{
+				CidrBlocks: container.ClusterMasterAuthorizedNetworksConfigCidrBlockArray{
+					&container.ClusterMasterAuthorizedNetworksConfigCidrBlockArgs{
+						CidrBlock:   pulumi.String("0.0.0.0/0"),
+						DisplayName: pulumi.String("All networks"),
+					},
+				},
+			},
+			Network:        gkeNetwork.Name,
+			NetworkingMode: pulumi.String("VPC_NATIVE"),
+			PrivateClusterConfig: &container.ClusterPrivateClusterConfigArgs{
+				EnablePrivateNodes:    pulumi.Bool(true),
+				EnablePrivateEndpoint: pulumi.Bool(false),
+				MasterIpv4CidrBlock:   pulumi.String("10.100.0.0/28"),
+			},
+			RemoveDefaultNodePool: pulumi.Bool(true),
+			ReleaseChannel: &container.ClusterReleaseChannelArgs{
+				Channel: pulumi.String("STABLE"),
+			},
+			Subnetwork: gkeSubnet.Name,
+			WorkloadIdentityConfig: &container.ClusterWorkloadIdentityConfigArgs{
+				WorkloadPool: pulumi.String(fmt.Sprintf("%v.svc.id.goog", gcpProject)),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a GCP Service Account for the node pool
+		gkeNodepoolSa, err := serviceaccount.NewAccount(ctx, "gke-nodepool-sa", &serviceaccount.AccountArgs{
+			AccountId:   pulumi.Sprintf("%v-np-1-sa", gkeCluster.Name),
+			DisplayName: pulumi.String("Nodepool 1 Service Account"),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a new node pool
+		_, err = container.NewNodePool(ctx, "gke-nodepool", &container.NodePoolArgs{
+			Cluster:   gkeCluster.ID(),
+			NodeCount: pulumi.Int(nodesPerZone),
+			NodeConfig: &container.NodePoolNodeConfigArgs{
+				OauthScopes: pulumi.StringArray{
+					pulumi.String("https://www.googleapis.com/auth/cloud-platform"),
+				},
+				ServiceAccount: gkeNodepoolSa.Email,
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Build Kubeconfig for accessing the cluster
+		clusterKubeconfig := pulumi.Sprintf(`apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: %[3]s
+    server: https://%[2]s
+  name: %[1]s
+contexts:
+- context:
+    cluster: %[1]s
+    user: %[1]s
+  name: %[1]s
+current-context: %[1]s
+kind: Config
+preferences: {}
+users:
+- name: %[1]s
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1beta1
+      command: gke-gcloud-auth-plugin
+      installHint: Install gke-gcloud-auth-plugin for use with kubectl by following
+        https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
+      provideClusterInfo: true
+        `, gkeCluster.Name, gkeCluster.Endpoint, gkeCluster.MasterAuth.ClusterCaCertificate().Elem())
+
+		// Export some values for use elsewhere
+		ctx.Export("networkName", gkeNetwork.Name)
+		ctx.Export("networkId", gkeNetwork.ID())
+		ctx.Export("clusterName", gkeCluster.Name)
+		ctx.Export("clusterId", gkeCluster.ID())
+		ctx.Export("kubeconfig", clusterKubeconfig)
```

Comment:
We're probably planning on mentioning this in the landing page, but figured I'd call it out just in case. My first attempt to output `kubeconfig` resulted in a file containing only `[secret]`. I had to pass `--show-secrets`:

```
pulumi stack output --show-secrets kubeconfig > kubeconfig
```

.. to get it to work.

Reply:
Out of curiosity; Does this only occur with A) the Go implementation and B) does the same level of configuration occur with Azure and AWS implementations?

Reply:
@timbohiatt AWS and Azure behave differently; the Kubeconfig doesn't have to be built manually, like we do here for GKE, and the export/output isn't treated like a secret. I'll dig into this code to see if I can figure out _why_ the Kubeconfig is being treated like a secret.

Reply:
Yeah this is actually what I thought you might say; I would be interested in your finding around this. I wonder if it's to do with the use how the GCP versions maybe uses the raw k8's connector and therefore requires the actual kubeconfig object to be built. We find this is a lot with GCP that typically we don't create "wrappers" around K8's standard content. I would be keen to see what you find out.

Reply:
I haven't gotten to the root cause of why the Go, Python, and TypeScript implementations treat the Kubeconfig as a secret, but for consistency's sake I've modified the YAML implementation so that it behaves similarly. This will at least give us consistency across languages within this set of templates. I'll push that change shortly.